### PR TITLE
WEBAPP-498: Refactor term extra

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

**Comment by Taggotty** [comment]
waits for https://github.com/Integreat/integreat-api-client/pull/41

**Comment by Taggotty** [comment]
> Quite some files changed which should not be modified:
> https://github.com/Integreat/integreat-webapp/pull/350/files#diff-9d7b2bde1dcf82f33ef3417334a260f0
> and lots of .csv and pathes of the locales. Please revert.

Isn't that because the local es branch hasn't been merged yet?

**Comment by klinzo** [comment]
> > Quite some files changed which should not be modified:
> > https://github.com/Integreat/integreat-webapp/pull/350/files#diff-9d7b2bde1dcf82f33ef3417334a260f0
> > and lots of .csv and pathes of the locales. Please revert.
> 
> Isn't that because the local es branch hasn't been merged yet? Can I merge the locales branch?

I don't think so, no. The PR in the locales repo does only modify the locales.json file and not any other files (e.g. csv, ...)

File: src/routes/offers/containers/OffersPage.js [hunk 1]
```diff
@@ -19,69 +19,69 @@ import CategoriesRouteConfig from '../../../modules/app/route-configs/Categories
...
 
 const mapStateToProps = (state: StateType) => ({
   city: state.location.payload.city,
   language: state.location.payload.language,
-  extraId: state.location.payload.extraId
+  offerId: state.location.payload.offerId
 })
 
 export default connect<*, *, *, *, *, *>(mapStateToProps)(
   withTranslation('extras')(
```
Comment (klinzo) [inline src/routes/offers/containers/OffersPage.js:85]: Wrong key. I guess we should check all other translations as well.

**Comment by klinzo** [comment]
Everything else looks fine, we should merge this asap in order to not get any more merge conflicts :)

File: src/routes/offers/containers/OffersPage.js (hunk 1 above)
Comment (maxammann) [inline src/routes/offers/containers/OffersPage.js:85]:
```suggestion
  withTranslation('offers')(
```

**Comment by maxammann** [comment]
2020-06-24-10:50:10-screenshot

**Comment by Taggotty** [comment]
> If I search for "extra" using intellij then I still find a lot of usages

The search in IntelliJ has some weird pagination. Those occurences weren't apparent the last time >:[
I hope I have removed the last occurences now, but feel free to checkout and check yourself.

**Pull request merged**